bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -688,13 +690,50 @@ public void deleteColumn(final String tableName, final String columnName) throws
   /** {@inheritDoc} */
   @Override
   public ClusterStatus getClusterStatus() throws IOException {
-    return new ClusterStatus() {
-      @Override
-      public Collection<ServerName> getServers() {
-        // TODO(sduskis): Point the server name to options.getServerName()
-        return Collections.emptyList();
-      }
-    };
+    try {
+      return new ClusterStatus() {
+        @Override
+        public Collection<ServerName> getServers() {
+          // TODO(sduskis): Point the server name to options.getServerName()
+          return Collections.emptyList();
+        }
+      };
+    } catch (NoSuchMethodError e) {
+      return getClusterStatusHbase2x();
+    }
+  }
+
+  private ClusterStatus getClusterStatusHbase2x() {
```

Comment:
is it possible to instead pull down and override the logic to the 2x implementation of BigtableAdmin?

Reply:
I don't quite understand what do you mean by saying "to pull down the logic".
The problem here that ClusterStatus 1x is dramatically different from ClusterStatus 2x. They don't even have the same constructor. So, I can't construct ClusterStatus 2x using API from 1x, the code won't compile.

Reply:
I agree with @kolea2, the there should be 2 separate implementations of this in bigtable-hbase1x and in bigtable-hbase2x.

Reply:
I got it, thanks for the explanation! Fixed as you suggested in a separate PR, so I close this one.